spark/src/main/scala/org/apache/spark/sql/delta/hooks/AutoCompact.scala

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.hooks
+
+import org.apache.spark.sql.delta.skipping.clustering.ClusteredTableUtils
+import org.apache.spark.sql.delta._
+import org.apache.spark.sql.delta.actions._
+import org.apache.spark.sql.delta.commands.{DeltaOptimizeContext, OptimizeExecutor}
+import org.apache.spark.sql.delta.commands.optimize._
+import org.apache.spark.sql.delta.metering.DeltaLogging
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+import org.apache.spark.sql.delta.stats.AutoCompactPartitionStats
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.catalog.CatalogTable
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.internal.SQLConf
+
+/**
+ * A trait for post commit hook which compacts files in a Delta table. This hook acts as a cheaper
+ * version of the OPTIMIZE command, by attempting to compact files to a smaller size.
+ *
+ * Auto Compact chooses files to compact greedily by looking at partition directories which
+ * have the largest number of files that are under a certain size threshold and launches a bounded
+ * number of optimize tasks based on the capacity of the cluster.
+ */
+trait AutoCompactBase extends PostCommitHook with DeltaLogging {
+
+  override val name: String = "Auto Compact"
```

Comment:
Compaction?

Reply:
I prefer "Auto Compact".. consistent with hook and feature name... not to be confused with optimize / compaction stuff which runs offline.